### PR TITLE
gnu-config: conan v2 support

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — DISMISSED** [review]

**Comment by SpaceIm** [comment]
```
Invalid configuration `x86_64-windows-msvc': OS `msvc' not recognized
gnu-config/cci.20201022 (test package): Current configuration is not supported by GNU config.
```

https://github.com/conan-io/conan/issues/7460

Maybe just skip this test if Visual Studio?

**Comment by SpaceIm** [comment]
```
'.' is not recognized as an internal or external command,
operable program or batch file.
gnu-config/cci.20201022 (test package): WARN: This conanfile has no build step
ERROR: gnu-config/cci.20201022 (test package): Error in test() method, line 24
	self.run("config.guess")
	ConanException: Error 1 while executing config.guess
```

in test_package, I don't understand what's wrong

**Review by jwillikers — DISMISSED** [review]

File: recipes/gnu-config/all/test_package/conanfile.py [hunk 1]
```diff
@@ -1,22 +1,28 @@
...
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
     def build_requirements(self):
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+        self.tool_requires(self.tested_reference_str)
+        if self._settings_build.os == "Windows" and \
+           not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
```
Comment (uilianries) [inline recipes/gnu-config/all/test_package/conanfile.py:19]:
```suggestion
           not self.conf.get("tools.microsoft.bash:path", default=False, check_type=str):
```

bash path is a string

**Review by jwillikers — APPROVED** [review]

**Comment by SpaceIm** [comment]
so conan 1.53.0 deployed now?

**Comment by SpaceIm** [comment]
@prince-chrismc conan 1.53.0 is not deployed.

**Comment by prince-chrismc** [comment]
#13753 is not in 🤫 well find out why from the team in Monday

File: recipes/gnu-config/all/conanfile.py [hunk 2]
```diff
@@ -41,14 +47,20 @@ def _extract_license(self):
...
 
     def package_info(self):
+        self.cpp_info.includedirs = []
+        self.cpp_info.libdirs = []
+
         bin_path = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH environment variable: {}".format(bin_path))
-        self.env_info.PATH.append(bin_path)
+        self.conf_info.define("user.gnu-config:config_guess", os.path.join(bin_path, "config.guess"))
+        self.conf_info.define("user.gnu-config:config_sub", os.path.join(bin_path, "config.sub"))
```
[the remark this answers is not in the document]
Comment (prince-chrismc) [reply recipes/gnu-config/all/conanfile.py:59]: Can't wait to see these in action 🙌

**Review by prince-chrismc — APPROVED** [review]

**Comment by SpaceIm** [comment]
> This PR has been triggered again because there are existing changes in `master` branch that are related to these same references resulting in different recipe-revisions. The build is being triggered on the PR context in case some additional fix is needed after merging those changes.

? no true

**Pull request merged**